zerver/migrations/0262_migrate_old_bot_messages.py

```diff
@@ -0,0 +1,64 @@
+# -*- coding: utf-8 -*-
+from __future__ import unicode_literals
+
+from django.conf import settings
+from django.db import migrations
+from django.db.backends.postgresql_psycopg2.schema import DatabaseSchemaEditor
+from django.db.migrations.state import StateApps
+
+from typing import Any
+
+def fix_messages(apps: StateApps, schema_editor: DatabaseSchemaEditor) -> None:
+    UserProfile = apps.get_model('zerver', 'UserProfile')
+    Message = apps.get_model('zerver', 'Message')
+    Realm = apps.get_model('zerver', 'Realm')
+
+    try:
+        internal_realm = Realm.objects.get(string_id=settings.SYSTEM_BOT_REALM)
+    except Realm.DoesNotExist:
+        # Server not initialized, or no system bot realm. Either way, we shouldn't do anything.
+        return
+
+    def get_bot_by_delivery_email(email: str) -> Any:
+        result = UserProfile.objects.select_related().filter(delivery_email__iexact=email.strip())
+        if not result:
+            raise UserProfile.DoesNotExist
+        if len(result) == 1:
+            return result[0]
+        else:
+            # If there's more than 1 account matching email, return the one in the system bot realm,
+            # or raise exception if doesn't exist.
+            return result.get(realm=internal_realm)
```

Comment:
Bot users will always have the same email and delivery email.  

And we should just add a `, realm=internal_realm` to the query filter and then not need all these conditionals -- `delivery_email` is a case-insensitive unique value within a given realm.

Reply:
So I think this function should just be `UserProfile.objects.select_related().get(delivery_email__iexact=email.strip(), realm=internal_realm)`.

Reply:
@timabbott That's how I initially did it, but then thought of this function to work around some weird configurations where users overrode the bot setting and it ended up existing in a different realm etc.
Limiting this to the internal realm would make testing simpler too, so that would be nice. Is it worth it to handle weird scenarios as mentioned above, or should I just test the simple approach and we'll go with that?

Reply:
If they changed the configuration to move those users into another realm, this migration probably isn't really required; the bot is probably just a normal bot within the realm where their users are.  So I think we can do the simple version.

Reply:
Ah, that sounds right.